lib/tag-list/input.tsx

```diff
@@ -1,49 +1,78 @@
-import React, { Component, FocusEvent, MouseEvent } from 'react';
+import React, {
+  ChangeEvent,
+  FocusEvent,
+  FunctionComponent,
+  useState,
+} from 'react';
+import { connect } from 'react-redux';
 import classNames from 'classnames';
 
+import actions from '../state/actions';
+
+import type * as S from '../state';
+import type * as T from '../types';
+
 type OwnProps = {
   editable: boolean;
   isSelected: boolean;
   onClick: (event: React.MouseEvent) => any;
-  onDone: (event: FocusEvent<HTMLInputElement>) => any;
-  value: string;
+  tagName: T.TagName;
 };
 
 type OwnState = {
   value: string;
 };
 
-type Props = OwnProps;
-
-export class TagListInput extends Component<Props, OwnState> {
-  constructor(props: Props) {
-    super(props);
-    this.state = { value: props.value };
-  }
-
-  render() {
-    const { editable, isSelected, onClick, onDone } = this.props;
-    const { value } = this.state;
-    const classes = classNames('tag-list-input', 'theme-color-fg', {
-      'is-selected': isSelected,
-    });
-
-    return editable ? (
-      <input
-        className={classes}
-        readOnly={!editable}
-        onClick={onClick}
-        value={value}
-        onChange={(e) => this.setState({ value: e.target.value })}
-        onBlur={onDone}
-        spellCheck={false}
-      />
-    ) : (
-      <button className={classes} onClick={onClick}>
-        {value}
-      </button>
-    );
-  }
-}
-
-export default TagListInput;
+type DispatchProps = {
+  renameTag: (oldTagName: T.TagName, newTagName: T.TagName) => any;
+};
+
+type Props = OwnProps & DispatchProps;
+
+export const TagListInput: FunctionComponent<Props> = ({
+  editable,
+  isSelected,
+  onClick,
+  renameTag,
+  tagName,
+}) => {
+  const [value, setValue] = useState(tagName);
```

Comment:
this here is a fairly appropriate place to use `useState`, but I find the name a bit generic. everything is a value. (again this was an existing issue so it's not particularly a problem with this PR, but this PR does offer a chance to clean it up, especially since we _are_ changing the way it's stored)

could we name it something meaningful like `tagName` and `setTagName` or `updateTagName`? maybe even `enteredTagName` or `partialTagName` or `tagFieldValue`?

Reply:
Good call! I've updated the naming for this now.